x-pack/plugins/upgrade_assistant/server/routes/app.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { API_BASE_PATH, DEPRECATION_LOGS_INDEX } from '../../common/constants';
+import { versionCheckHandlerWrapper } from '../lib/es_version_precheck';
+import { Privileges } from '../shared_imports';
+import { RouteDependencies } from '../types';
+
+const extractMissingPrivileges = (
+  privilegesObject: { [key: string]: Record<string, boolean> } = {}
+): string[] =>
+  Object.keys(privilegesObject).reduce((privileges: string[], privilegeName: string): string[] => {
+    if (Object.values(privilegesObject[privilegeName]).some((e) => !e)) {
+      privileges.push(privilegeName);
+    }
+    return privileges;
+  }, []);
+
+export function registerAppRoutes({ router, lib: { handleEsError } }: RouteDependencies) {
+  router.get(
+    {
+      path: `${API_BASE_PATH}/privileges`,
+      validate: false,
+    },
+    versionCheckHandlerWrapper(
+      async (
+        {
+          core: {
+            elasticsearch: { client },
+          },
+        },
+        request,
+        response
+      ) => {
+        const privilegesResult: Privileges = {
+          hasAllPrivileges: true,
+          missingPrivileges: {
+            index: [],
+          },
+        };
+
```

Comment:
I saw in other apps we also check for `isSeecurityEnabled()` and based on that we let the users use the app anyway or not. I wasn't sure that was something we'd also want to do here or not.

Reply:
Yes if security isn't enabled then we should let the user proceed as if they have all required privileges.